### PR TITLE
DROID-2686 App | Tech | AwaitAccountStart refactoring - get rid of boolean, introduce explicit state

### PR DESCRIPTION
**Pull request by uburoiubu**

### CODE REVIEW

File: domain/src/main/java/com/anytypeio/anytype/domain/multiplayer/ActiveSpaceMemberSubscriptionContainer.kt [hunk 1]
```diff
@@ -43,11 +43,18 @@ interface ActiveSpaceMemberSubscriptionContainer {
...
             scope.launch {
-                awaitAccountStart.isStarted().collect { isStarted ->
-                    if (isStarted)
-                        start()
-                    else
-                        stop()
+                awaitAccountStart.state().collect { state ->
+                    when(state) {
+                        AwaitAccountStartManager.State.Init -> {
+                            // Do nothing
```
Comment (konstantiniiv) [inline domain/src/main/java/com/anytypeio/anytype/domain/multiplayer/ActiveSpaceMemberSubscriptionContainer.kt:49]: Timber.i

File: domain/src/main/java/com/anytypeio/anytype/domain/multiplayer/ActiveSpaceMemberSubscriptionContainer.kt [hunk 2]
```diff
@@ -43,11 +43,18 @@ interface ActiveSpaceMemberSubscriptionContainer {
...
+                        }
+                        AwaitAccountStartManager.State.Started -> {
+                            start()
```
Comment (konstantiniiv) [inline domain/src/main/java/com/anytypeio/anytype/domain/multiplayer/ActiveSpaceMemberSubscriptionContainer.kt:52]: Timber.i

File: domain/src/main/java/com/anytypeio/anytype/domain/multiplayer/SpaceViewSubscriptionContainer.kt [hunk 3]
```diff
@@ -51,11 +51,18 @@ interface SpaceViewSubscriptionContainer {
...
             scope.launch {
-                awaitAccountStart.isStarted().collect { isStarted ->
-                    if (isStarted)
-                        start()
-                    else
-                        stop()
+                awaitAccountStart.state().collect { state ->
+                    when(state) {
+                        AwaitAccountStartManager.State.Init -> {
+                            // Do nothing
```
Comment (konstantiniiv) [inline domain/src/main/java/com/anytypeio/anytype/domain/multiplayer/SpaceViewSubscriptionContainer.kt:57]: Timber.i

**Review by konstantiniiv — APPROVED** [review]

File: domain/src/main/java/com/anytypeio/anytype/domain/multiplayer/ActiveSpaceMemberSubscriptionContainer.kt (hunk 1 above)
Comment (uburoiubu) [reply domain/src/main/java/com/anytypeio/anytype/domain/multiplayer/ActiveSpaceMemberSubscriptionContainer.kt:49]: will be added in the next PR

File: domain/src/main/java/com/anytypeio/anytype/domain/multiplayer/ActiveSpaceMemberSubscriptionContainer.kt (hunk 2 above)
Comment (uburoiubu) [reply domain/src/main/java/com/anytypeio/anytype/domain/multiplayer/ActiveSpaceMemberSubscriptionContainer.kt:52]: idem

File: domain/src/main/java/com/anytypeio/anytype/domain/multiplayer/SpaceViewSubscriptionContainer.kt (hunk 3 above)
Comment (uburoiubu) [reply domain/src/main/java/com/anytypeio/anytype/domain/multiplayer/SpaceViewSubscriptionContainer.kt:57]: idem

**Pull request merged**